src/YoutubeAtom.tsx

```diff
@@ -89,42 +255,55 @@ export const YoutubeAtom = ({
     const playerReadyCallback = useCallback(() => setPlayerReady(true), []);
 
     return (
-        <MaintainAspectRatio height={height} width={width}>
-            {loadPlayer && consentState && (
-                <YoutubeAtomPlayer
-                    videoId={assetId}
-                    overrideImage={overrideImage}
-                    posterImage={posterImage}
-                    adTargeting={adTargeting}
-                    consentState={consentState}
-                    height={height}
-                    width={width}
-                    title={title}
-                    origin={origin}
-                    eventEmitters={eventEmitters}
-                    /**
-                     * If there is an overlay we want to autoplay
-                     * If there is not an overlay the user will use the YouTube player UI to play
-                     */
-                    autoPlay={hasOverlay}
-                    onReady={playerReadyCallback}
-                />
-            )}
-            {showOverlay && (
-                <YoutubeAtomOverlay
-                    videoId={assetId}
-                    overrideImage={overrideImage}
-                    posterImage={posterImage}
-                    height={height}
-                    width={width}
-                    alt={alt}
-                    role={role}
-                    duration={duration}
-                    pillar={pillar}
-                    onClick={() => setOverlayClicked(true)}
-                />
-            )}
-            {showPlaceholder && <YoutubeAtomPlaceholder videoId={assetId} />}
-        </MaintainAspectRatio>
+        <div ref={setRef} css={isSticky && stickyContainerStyles(192)}>
```

Comment:
This is a non-blocking thought on component composition.

I can see how moving sticky behaviour to the atom has simplified things from the app's pov.

However moving this behaviour I feel has mixed sticky UI and state concerns with the coordination role that `YoutubeAtom` used to do.

Previously `YoutubeAtom` separated concerns of coordination (the parent), UI (overlay and placeholder) and the player itself.

Is there scope to have a structure something like:

```js
<YoutubeAtomSticky sticky={sticky} onStopVideo={() => setStopVideo(true)}>
    <MaintainAspectRatio ...>
        <YoutubeAtomPlayer stopVideo={stopVideo} />
        <YoutubeAtomOverlay ...>
        <YoutubeAtomPlaceholder ...>
    </MaintainAspectRatio>
</YoutubeAtomSticky>
```

Where `YoutubeAtomSticky` encapsulate the internal sticky UI and state?

This could be a follow up and just a thought for consideration :)

Reply:
If this change works and doesn't add any complexity then I think this makes a lot of sense. I find the composition approach here a lot easier to approach. I'm aware though that sometimes these sorts of things don't work out as well as they might so I'm happy to leave the choice with @joecowton1

Reply:
Yeah I could totally have missed something which means this is not practical. Thanks for taking a look @joecowton1

Reply:
That worked out nicely in the end 👌